department-of-veterans-affairs/abd-vro
# MCP-2116 - Fix status code for missing collectionId field in requestBody

## What was the problem?

When POSTing to `/v2/examOrderingStatus` endpoint:

- If collectionId field is missing, the response status code is 200 OK but the expected result  status code should be 400 Bad Request for missing required parameter.

Associated tickets or Slack threads:

- [MCP-2116](https://amida.atlassian.net/browse/MCP-2116)

## How does this fix it?[^1]

The previous Java field was using the `int` primitive with a `NotNull` validation. However, since `int` cannot be null, we must declare the field as `Integer` (which can be null) to account for this scenario.

## How to test this PR
- Step 1: Submit a POST request to `/v2/examOrderingStatus` with a payload that omits `collectionId`
  - Verify you receive a `400 Bad Request` with a message indicating that the field is required
- Step 2: Submit a valid POST request to `/v2/examOrderingStatus` with a valid payload
  - Verify you receive a `200 OK`



[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
